plugins/de.cognicrypt.codegenerator/src/main/java/de/cognicrypt/codegenerator/generator/CrySLBasedCodeGenerator.java

```diff
@@ -683,11 +683,11 @@ private CrySLMethod fetchCorrespondingMethod(Entry<CrySLPredicate, Entry<CrySLRu
 			} else {
 				try {
 					methodParameter[i] = Class.forName(parameter.getValue());
-					i++;
 				} catch (ClassNotFoundException e) {
 					Activator.getDefault().logError(e, "No class found for type: " + parameter.getValue().toString());
 				}
 			}
+			i++;

```

Comment:
Why did you move this out of the try-block?

Reply:
This is a bug. `i` never gets incremented for primitive type parameters.